### PR TITLE
Add settings to Hammer

### PR DESCRIPTION
**Pull request by pieter-edelman-nictiz**
As there doesn's seem to be a good cross-platform settings library for .NET, the QML Settings object is used; although it's somewhat unstable, it is based on the very robust QSettings library. However, we have to be, er, 'creative' to make it work in Qml.Net. I'd be happy to hear if you know a better way though ...

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -952,6 +952,9 @@ static int Main(string[] args)
           return 1;
         }
 
+        // This is a hack, see the comment in the file 
+        engine.Load("ApplicationName.qml");
```
Comment (vadi2) [inline Program.cs]:
```suggestion
        QCoreApplication.OrganizationDomain = "Hammer.mc";
        QCoreApplication.OrganizationName = "Hammer";
```

This seems to work fine (see https://github.com/qmlnet/qmlnet/issues/133)

File: Main.qml [hunk 2]
```diff
@@ -5,16 +5,19 @@ import QtQuick.Controls.Material 2.12
...
 import QtQuick.Controls.Universal 2.12
 import QtQuick.Layouts 1.12
+import QtQml 2.12
 import Qt.labs.platform 1.1
+import Qt.labs.settings 1.0
 
 ApplicationWindow {
     id: window
     visible: true
+    visibility: settings.windowVisibility
```
Comment (vadi2) [inline Main.qml]: Are you sure we want to save this? When you launch a program, it should always show, even if it was minimized on exit

File: Program.cs (hunk 1 above)
Comment (pieter-edelman-nictiz) [reply Program.cs]: Awesome, much better!

File: Main.qml (hunk 2 above)
Comment (pieter-edelman-nictiz) [reply Main.qml]: This is actually about maximized/windowed state. Maybe it's better to just save an isMaximized bool.

Comment (vadi2) [reply Main.qml]: That makes sense.

**Pull request merged**